src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -573,6 +576,9 @@ def build_vmss_resource(name, naming_prefix, location, tags, overprovision, upgr
     if custom_data:
         os_profile['customData'] = b64encode(custom_data)
 
+    if secrets:
+        os_profile['secrets'] = secrets
```

Comment:
Rather than have a "vm-format" command, why can you not just do the formatting here through a helper function or within the template_builder.py file for VM/VMSS?  There are things such as "build_storage_profile" and so on that essentially fill in those missing pieces.